pkg/scheduler/eventhandlers.go

```diff
@@ -523,23 +523,44 @@ func addAllEventHandlers(
 }
 
 func nodeSchedulingPropertiesChange(newNode *v1.Node, oldNode *v1.Node) *framework.ClusterEvent {
+	var action framework.ActionType
+	var label string
+	var count int
+
 	if nodeSpecUnschedulableChanged(newNode, oldNode) {
-		return &queue.NodeSpecUnschedulableChange
+		action |= framework.UpdateNodeTaint
+		label = queue.NodeSpecUnschedulableChange.Label
+		count++
 	}
 	if nodeAllocatableChanged(newNode, oldNode) {
-		return &queue.NodeAllocatableChange
+		action |= framework.UpdateNodeAllocatable
+		label = queue.NodeAllocatableChange.Label
+		count++
 	}
 	if nodeLabelsChanged(newNode, oldNode) {
-		return &queue.NodeLabelChange
+		action |= framework.UpdateNodeLabel
+		label = queue.NodeLabelChange.Label
+		count++
 	}
 	if nodeTaintsChanged(newNode, oldNode) {
-		return &queue.NodeTaintChange
+		action |= framework.UpdateNodeTaint
+		label = queue.NodeTaintChange.Label
+		count++
 	}
 	if nodeConditionsChanged(newNode, oldNode) {
-		return &queue.NodeConditionChange
+		action |= framework.UpdateNodeCondition
+		label = queue.NodeConditionChange.Label
+		count++
 	}
 
-	return nil
+	if count == 0 {
+		return nil
+	}
+
+	if count > 1 {
+		label = queue.NodeUpdate.Label
```

Comment:
Very minor thing:

Seems this `Label` is used in logs + metrics. 
Meaning, if we go with this implementation, the label value of metrics could get changed in these two scenarios:
- Only Node's label is updated → metrics will get NodeLabelChange label.
- Both Node's label and taint are updated in a single update operation → metrics will get NodeUpdated label.

Isn't that a bit weird? Maybe we can just always return NodeUpdate for all node related events to unite the label value in those kind of scenarios.

Reply:
> Isn't that a bit weird? since NodeLabelChange label in metrics is actually not enough for catching all Node's label updates.

Sorry, can't quite follow you. Is there anything wrong with nodeLabelChange?

Reply:
Let's say you want to check "How many times the events of Node label change have moved the Pods to activeQ ([ref](https://github.com/kubernetes/kubernetes/blob/d61cbac69aae97db1839bd2e0e86d68f26b353a7/pkg/scheduler/internal/queue/scheduling_queue.go#L1129))". 
You cannot simply query an accurate value with `event: NodeLabelChange` because `event: NodeUpdated` might include NodeLabelChange actually. (since if, for example, both label and taint are updated in a single node update operation, metrics will get NodeUpdated label, not NodeLabelChange.)

Reply:
Make sense to me. Let's only fix the bug here and make the count accurate, fall back to https://github.com/kubernetes/kubernetes/pull/118490 but with more tests here.

Reply:
Done @sanposhiho

Reply:
/hold

For another discussion

Reply:
I would not hope to change the behavior, unless that's necessary. And I think we can save little mem here unless many pods update with more than one change, like label and annotation the same time.

Reply:
> I think we can save little mem here unless many pods update with more than one change

OK, fair point. I couldn't tell the actual case where one part and another in Node is updated at the same time in real life. So, let's just keep it in mind, until we get some evidence (e.g., hoge-controller updates Node's A and B at the same time.).

/unhold